hs_app_netCDF/models.py

```diff
@@ -75,38 +67,26 @@ def create(cls, **kwargs):
 
     @classmethod
     def update(cls, element_id, **kwargs):
-        ori_cov = OriginalCoverage.objects.get(id=element_id)
-        if ori_cov:
-            # update bounding box info
-            if 'value' in kwargs:
-                if not isinstance(kwargs['value'], dict):
-                    raise ValidationError('Invalid coverage value format.')
+        try:
+            ori_cov = OriginalCoverage.objects.get(id=element_id)
+        except ObjectDoesNotExist:
+            raise ObjectDoesNotExist("No coverage element was found for the provided id:%s" % element_id)
 
-                value_dict = ori_cov.value
+        # update bounding box info
+        if 'value' in kwargs:
+            if not isinstance(kwargs['value'], dict):
+                raise ValidationError('Invalid coverage value format.')
```

Comment:
type checking not recommended. documented accepting parameters is a better idea.